staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -250,6 +250,12 @@ type ObjectMeta struct {
 	// is an identifier for the responsible component that will remove the entry
 	// from the list. If the deletionTimestamp of the object is non-nil, entries
 	// in this list can only be removed.
+	// Finalizers may be processed and removed in any order.  Order is NOT enforced
+	// because it introduces significant risk of stuck finalizers.
+	// finalizers is a shared field, any actor with permission can reorder it.
+	// If the finalizer list is processed in order, then this can deadlock finalizers.
```

Comment:
"then this can deadlock finalizers" makes it sound like a bug in the API server. Maybe rephrase like "this can lead to a situation in which the component responsible for the first finalizer in the list is waiting for a condition produced by a component responsible for a finalizer later in the list, resulting in a deadlock."

Reply:
updated